### ISSUE TITLE
[HIGH] Fix validation naive accuracy

### ISSUE THREAD

**Issue by g-simmons** [g-simmons/289G_NLP_project_FQ2020#37]
<img width="664" alt="image" src="https://user-images.githubusercontent.com/28633482/102209157-3377e500-3e85-11eb-959e-414365069470.png">

Above plot shows our training accuracy, training accuracy that you would get from predicting all-negative (naive) for each sample, our validation accuracy (calculated at every 0.25 epoch), and the green line that is barely on the plot is naive validation accuracy. 

First of all congratulations everyone, the model learned something over baseline 👏 , and this is without BERT

Second of all, I believe the naive validation accuracy has issues due to batch size. 

Background - The data points for the validation metrics both naive and predicted are averaged across the entire validation set. This is in contrast to the training metrics that are just plotted directly at each step. (The reason for this was difficulty in synchronizing the plots). Additionally, batch size for training is 16, batch size for validation is 1. 

What I think is wrong - I am imagining a scenario where we have some examples with an unusually high proportion of positives, and probably a low number of total candidates. For these samples, the all-negative baseline will do poorly, but if these samples were part of a batch the effect would be concealed. Since we are averaging the results for single samples rather than batches for val, this drags down the baseline accuracy. 

Some solutions:
- Establish a scalar baseline average acc across the entire dataset and plot as horizonal line
- increase val batch size
-...?

Some questions:
- In general, should we concat first and then average, or average and then average again?
